profiles-sdk-examples/docs/catalog.md

```diff
@@ -0,0 +1,352 @@
+# Cortex Catalog
+
+> The Cortex Catalog is a registry for Agents, Skills, Types, Connections, Data Sources, Profiles, and Campaigns.
+(From: https://cognitivescale.github.io/cortex-fabric/docs/reference-guides/cortex-service-descriptions)
+
+The Profiles SDK includes a client for the Cortex Catalog that utilizes
+the [Cortex Fabric GraphQL API](https://cognitivescale.github.io/cortex-fabric/docs/reference-guides/apis/#graphql-api).
+Deployed Agents interacts with the Cortex Catalog instance. The Profiles SDK includes a local
+implementation of the Catalog that may be easier to use during development.
+
+Base Interface: `com.c12e.cortex.phoenix.Catalog` <!-- TODO: Link to javadoc -->
+
+## Remote Catalog
+
+By default, the Profiles SDK (`CortexSession`) utilizes remote Cortex Catalog. This is such that deployed Agents
+containing your application (and the Cortex Profiles SDK) communicates with the Cortex Catalog instance in your
+Cortex Cluster.
+
+Implementation: `com.c12e.cortex.profiles.catalog.CortexRemoteCatalog` <!-- TODO: Link to javadoc -->
+
+### Configuration Properties
+
+The `CortexRemoteCatalog` implementation requires:
+* The Cortex API endpoint to be specified (`https://api.<dci-base-domain>`) - `spark.cortex.client.phoenix.url`
+* A Cortex API Token - `spark.cortex.client.phoenix.token`
+
+When packaged as a Skill with the existing templates then token and URL are handled by the Cortex Cluster. These
+configuration options need only be set when running outside the Cortex cluster (i.e. locally).
+
+**WARNING**: The `CortexRemoteCatalog` implementation is being developed and not all methods (GraphQL endpoints) are
+currently supported. Unsupported methods will return a `NotImplementedError` or `RuntimeException` error.
+
+## Local Catalog
+
+The `LocalCatalog` provides a file-based implementation for mocking the actual Cortex Catalog. This can be useful while developing
+your solution because
+- Cortex Resources (Data Sources, Connections, and Profile Schemas) can be written locally in YAML.
+- Smaller datasets can be used for Data Sources and Connections before actual deployment.
+- Local files can be used for datasets, which need not require Secrets or authentication in general. (NOTE: Remote connections are supported but require setting up a secrets client).
```

Comment:
NOTE should not be in parenthesis. give it it's own line and bold the word NOTE.